hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.hive;
+
+import java.util.Collections;
+import java.util.Map;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.TableType;
+import org.apache.hadoop.hive.metastore.api.SerDeInfo;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.iceberg.BaseMetastoreTableOperations;
+import org.apache.iceberg.ClientPool;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.exceptions.NoSuchIcebergTableException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** All the HMS operations like table,view,materialized_view should implement this. */
+interface HiveOperationsBase {
+
+  Logger LOG = LoggerFactory.getLogger(HiveOperationsBase.class);
+  // The max size is based on HMS backend database. For Hive versions below 2.3, the max table
+  // parameter size is 4000
+  // characters, see https://issues.apache.org/jira/browse/HIVE-12274
+  // set to 0 to not expose Iceberg metadata in HMS Table properties.
+  String HIVE_TABLE_PROPERTY_MAX_SIZE = "iceberg.hive.table-property-max-size";
+  long HIVE_TABLE_PROPERTY_MAX_SIZE_DEFAULT = 32672;
+
+  TableType tableType();
+
+  ClientPool<IMetaStoreClient, TException> metaClients();
+
+  long maxHiveTablePropertySize();
+
+  String database();
+
+  String table();
+
+  Map<String, String> hmsEnvContext(String metadataLocation);
```

Comment:
How will this work for View-s?
Any thoughts here?

Reply:
In case of `Table` envContext is looking like [here](https://github.com/apache/iceberg/blob/2e2ac8dfd2ab2c214444cccb0e6024e1f80b0502/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L333) . 
`expectedMetadataLocation != null
                    ? ImmutableMap.of(
                        NO_LOCK_EXPECTED_KEY,
                        METADATA_LOCATION_PROP,
                        NO_LOCK_EXPECTED_VALUE,
                        expectedMetadataLocation)
                    : ImmutableMap.of()`

In case of `View` it will look [like](https://github.com/apache/iceberg/pull/8907/files#diff-db46657b84d66e084e15f31b8dab21577efb2ae7102863f94c6c9477782de676R283) 
`expectedMetadataLocation != null
                    ? ImmutableMap.of(
                        BaseMetastoreTableOperations.METADATA_LOCATION_PROP,
                        expectedMetadataLocation)
                    : ImmutableMap.of()`